### ISSUE TITLE
Ammunition consumables can cause NPE if improperly declared

### ISSUE THREAD

**Issue by draconas1** [EndlesNights/dnd4eBeta#124]
Moving this to an issue because the git commit message was turning into a saga.

Consumables come in 3 parts: type, target and quantity.  Consumables can be used by both the power, and the weapon used by the power.  

The consumable check would trigger off type but would then assume that target existed and could be found, and that quantity existed and was a number.  It was entirely possible to only declare the type and leave the others blank.  As happened in Steve, the example character.  This causes a null pointer exception when trying to find the ammunition details which locks up that attack roll forever.  

Also because it performed an addition onto existing data rather than an overwrite, the weapon consumable implementation would continue incrementing the ammo attack bonus if the attack was rolled repeatedly due to persisted state.   The power ammo bonus was an overwrite so did not have this problem.  

Finally the code blocks for the 2 were very similar, with a large setup block, and only small differences in the final implementation.

**Comment by draconas1** [comment]
Solution:  
Inner functions.  I needed to use an inner function as the check structure and the implementation of the consumable powers modify state in the wider method, so need to have access to all of it.

Factored 3 inner functions: 1 for the power ammunition use effect.  One for the weapon ammunition use effect.  One for the overall control structure.  
Fixed the overall control structure such that it copes with dodgy input data.  In the case it hits an error it now drops a notification warning to the UI.  
I choose warning because from reading the code I assume that failing to get this consumable is not a critical failure that should block the power use.  I did however want to let the user know what was happening because it implies something is wrong with their power.  

To deal with the maintained state I set rollData["ammo] = 0 as soon as the rolldata has been acquired, so I can increment freely.
